Source/WebCore/dom/EventLoop.cpp

```diff
@@ -488,11 +545,68 @@ void EventLoopTaskGroup::queueTask(TaskSource source, EventLoop::TaskFunction&&
     return queueTask(makeUnique<EventLoopFunctionDispatchTask>(source, *this, WTFMove(function)));
 }
 
+class EventLoopFunctionMicrotaskDispatcher final : public JSC::MicrotaskDispatcher {
+    WTF_MAKE_TZONE_ALLOCATED(EventLoopFunctionMicrotaskDispatcher);
+public:
+
+    EventLoopFunctionMicrotaskDispatcher(EventLoopTaskGroup& group, EventLoop::TaskFunction&& function)
+        : m_group(group)
+        , m_function(WTFMove(function))
+    {
+    }
+
+    ~EventLoopFunctionMicrotaskDispatcher() final = default;
+
+    JSC::QueuedTask::Status run(JSC::QueuedTask&) final
+    {
+        auto group = m_group.get();
+        if (!group || group->isStoppedPermanently())
+            return JSC::QueuedTask::Status::Stopped;
+        if (group->isSuspended())
+            return JSC::QueuedTask::Status::Suspended;
+
+        m_function();
+        return JSC::QueuedTask::Status::Executed;
+    }
+
+    bool isRunnable() final
+    {
+        auto group = m_group.get();
+        return group && !group->isStoppedPermanently() && !group->isSuspended();
+    }
+
+    static Ref<EventLoopFunctionMicrotaskDispatcher> create(EventLoopTaskGroup& group, EventLoop::TaskFunction&& function)
+    {
+        return adoptRef(*new EventLoopFunctionMicrotaskDispatcher(group, WTFMove(function)));
+    }
+
+private:
+    WeakPtr<EventLoopTaskGroup> m_group;
+    EventLoop::TaskFunction m_function;
+};
+
+WTF_MAKE_TZONE_ALLOCATED_IMPL(EventLoopFunctionMicrotaskDispatcher);
+
 void EventLoopTaskGroup::queueMicrotask(EventLoop::TaskFunction&& function)
+{
+    queueMicrotask(JSC::QueuedTask {
+        EventLoopFunctionMicrotaskDispatcher::create(*this, WTFMove(function)),
+        JSC::MicrotaskIdentifier::generate(),
```

Comment:
Can we add a `QueuedTask` constructor which takes two arguments instead of setting all these arguments to `nullptr` or `{ }`?